pkg/statsd/statsd_defaults_and_params.go

```diff
@@ -32,6 +35,43 @@ var DefaultTags = gostatsd.Tags{}
 // DefaultInternalTags is the default list of additional tags on internal metrics
 var DefaultInternalTags = gostatsd.Tags{}
 
+// DefaultCloudProviderCacheValues contains the default cache values for each cloud provider.
+var DefaultCloudProviderCacheValues = map[string]CacheOptions{
+	k8s.ProviderName: {
+		// Basically the CacheRefreshPeriod is the granularity at which we care about batching data.
+		// If 5 requests come in inside 50 milliseconds, and this is set to 50 milliseconds, we only need to look
+		// up the informer once, and do the regexes once. This saves us resources at the cost of data precision.
+		CacheRefreshPeriod:        50 * time.Millisecond,
```

Comment:
There's a `time.Ticker` running at this interval, is that acceptable? It will also flush the entire CP cache every time, effectively disabling it, but also creating memory churn and garbage as it tries to repopulate it. This is fine on a local instance, but may hurt something which monitors an entire cluster.

The batching is in `lookupDispatcher`, and it waits for more IPs for 10ms (hard coded) after the first IP comes in, then processes what it has.  Generally the CP caching and lookup behavior can be pretty tough to follow because of all the batching and async stuff :(

To be clear:
- `CacheRefreshPeriod`: how often the `CloudHandler` will check if entries should be evicted or refreshed
- `CacheEvictAfterIdlePeriod`: when the refresh runs, if an entry hasn't been accessed in this interval, evict it
- `Cache[Negative]TTL`: if an entry is kept, and it's been this long since it was last created/refreshed (**not** last time the cached value was accessed), refresh it from the backing CP.

Reply:
Yeah I had real trouble diving into this codebase in general due to all the async stuff. I think I've done OK but I'm really relying on you and Mikhail to pick out any weird edge cases or performance things for me since I don't have a lot of context.

Perhaps negative TTL and CacheRefreshPeriod are the things I need to tweak here then? Something like `CacheRefreshPeriod=1s`, `CacheEvictAfterIdlePeriod=60s`, `CacheNegativeTTL=1s`. This would keep everything refreshed and evict things after a while.

I might make a test to try and get the situation to happen at AWS cache values and then try and fix it with more aggressive cache values to prove the fix works.

We can probably deal with metrics not having sub second granularity if a pod somehow manages to stop and another one start with the same IP immediately (very rare edge case).

WDYT?

Reply:
I think a low-ish (15 second) `CacheRefreshPeriod` is fine, keep `CacheEvictAfterIdlePeriod` at default (DefaultCacheEvictAfterIdlePeriod = 10 min), and keep the TTLs at 1ms (or even 0ms).

Since there's no network cost to doing a refresh, and we're not refreshing multiple times a second, it should be fine to refresh the cache every time, even though there's a bit of a hit on the CPU.  I doubt it would be noticeable.